Gifski/Utilities.swift

```diff
@@ -758,6 +787,63 @@ extension AVAssetTrack {
 	}
 }
 
+extension AVAssetTrack {
+	/**
+	Removes blank frames from the begining of the track.
+
+	This can be useful to trim blank frames from files produced by tools like the iOS simulator screen recorder.
+	*/
+	func trimBlankFrames() throws -> AVAssetTrack {
+		guard let asset = self.asset else {
+			throw VideoTrimmingError.assetIsNil
+		}
+
+		let reader = try AVAssetReader(asset: asset)
+
+		let readerOutput = AVAssetReaderTrackOutput(track: self, outputSettings: nil)
+		reader.add(readerOutput)
+		reader.startReading()
+
+		while let sampleBuffer = readerOutput.copyNextSampleBuffer() {
+			guard reader.status == .reading else {
+				reader.cancelReading()
+				throw VideoTrimmingError.assetReaderFailure
+			}
+
+			// On first non-empty frame
+			guard sampleBuffer.totalSampleSize == 0 else {
+				let currentTimestamp = sampleBuffer.outputPresentationTimeStamp
+
+				// Add video track to composition and remove time range (from the begining up to current frame)
+				let composition = AVMutableComposition()
+				guard let videoCompTrack = composition.addMutableTrack(withMediaType: .video, preferredTrackID: CMPersistentTrackID()) else {
+					reader.cancelReading()
+					throw VideoTrimmingError.compositionCouldNotBeCreated
+				}
+
+				do {
+					try videoCompTrack.insertTimeRange(.init(start: .zero, duration: asset.duration), of: self, at: .zero)
+				} catch {
+					reader.cancelReading()
+					throw error
+				}
+
+				videoCompTrack.removeTimeRange(CMTimeRange(start: .zero, end: currentTimestamp))
+
+				reader.cancelReading()
```

Comment:
I think you can just put this inside a `defer` at the top of the scope.

Reply:
Fixed in 50532fc.